app/src/main/java/pm/gnosis/heimdall/ui/safe/details/SafeDetailsFragment.kt

```diff
@@ -62,8 +61,11 @@ class SafeDetailsFragment : BaseFragment() {
         safeAddress = arguments?.getString(EXTRA_SAFE_ADDRESS)?.asEthereumAddress()!!
         viewModel.setup(safeAddress)
 
-        layout_safe_details_send_button.setCompoundDrawableResource(left = R.drawable.ic_send_azure)
-        layout_safe_details_receive_button.setCompoundDrawableResource(left = R.drawable.ic_qrcode_scan_azure)
+        layout_safe_details_safe_image.setAddress(safeAddress)
+
+        val addressString = safeAddress.asEthereumAddressChecksumString()
```

Comment:
Not sure if I would do this on the UI thread, since this contains hashing and on older devices might be slow ... not sure how big of an issue it really could be

Reply:
Calculation moved to worker thread